src/cpu/o3/BaseO3CPU.py

```
@@ -251,3 +251,48 @@ def support_take_over(cls):
 
     enableConstantFolding = Param.Bool(False, "Enable Constant Folding (add-immediate elimination)")
     enableMovImmElimination = Param.Bool(False, "Enable MOVI elimination")
+
+    # Trace mode parameters for trace-driven simulation
+    enableTraceMode = Param.Bool(False, "Enable trace-driven simulation mode")
```

Ruff (0.14.10)

`Param` may be undefined, or defined from star imports

(F405)